src/js/modules/public/Utils.ts

```diff
@@ -133,11 +133,18 @@ export const calculateRecordLength = (record: Record): number => {
   size += varintZigzagSize(BigInt(record.valueLen));
   size += record.value.length;
   size += varintZigzagSize(BigInt(record.headers.length));
-  size += varintZigzagSize(BigInt(size));
   return size;
 };
 
 export const calculateRecordBatchSize = (records: Record[]): number => {
-  // 61 is the header batch bytes size
-  return 61 + records.reduce((p, r) => p + r.length, 0);
+  // 57 is the header batch bytes size
+  // 4 is the bytes uses for array size
+  return (
+    57 +
```

Comment:
prefer named constants over magic numbers

Reply:
Got it, sounds good

Reply:
Done